packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -0,0 +1,220 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import ssm = require('./ssm.generated');
+
+/**
+ * An SSM Parameter reference.
+ */
+export interface IParameter extends cdk.IConstruct {
+  /**
+   * The ARN of the SSM Parameter resource.
+   */
+  readonly parameterArn: string;
+
+  /**
+   * The name of the SSM Parameter resource.
+   */
+  readonly parameterName: string;
+
+  /**
+   * The type of the SSM Parameter resource.
+   */
+  readonly parameterType: string;
+
+  /**
+   * Grants read (DescribeParameter, GetParameter, GetParameterHistory) permissions on the SSM Parameter.
+   *
+   * @param grantee the role to be granted read-only access to the parameter.
+   */
+  grantRead(grantee: iam.IRole): IParameter;
```

Comment:
I think we don't do fluent interfaces if we can avoid them.

Reply:
Agreed, no fluent

Reply:
`grantee` needs to be `iam.IPrincipal`